### ISSUE TITLE
EKF and UKF classes incompatible with scalar measurements

### ISSUE THREAD

**Issue by lundalana** [rlabbe/Kalman-and-Bayesian-Filters-in-Python#366]
The particular application I'm working uses a single measurement at each time step to update the Kalman Filter. I noticed that in both the UKF and EKF classes the functions used in designing the update inhibit scalar measurements from being used effectively. Both return errors concerning matrix size.

**Comment by rlabbe** [comment]
scalars make matrix multiplication difficult; when I started this project I made the linear kalman filter able to accept scalars, but it is just a pain so I abandoned that in the subsequent EKF/UKF.

Just wrap the measurement in an array:

    z = np.array([[z]])

Likewise, your measurement noise cannot be a scalar, but an array

    R = np.array([[1.2]]) # or whatever

not

    R = 1.2

If I remember correctly UKF wants a 1d measurement array? If so, then:

    z = np.array([z])

If I misunderstood what you are trying to do let me know, but as I understand the question this is as designed.

**Comment by o-alexandre-felipe** [comment]
I just experienced the same problem, I looked into the code and could fix this issue alone.

However the discussion on https://github.com/rlabbe/filterpy/issues/179, makes me think that maybe It needs a closer look to solve multiple potential issues at once.

**Comment by lundalana** [comment]
Thank you both for your comments and help on this problem. I agree that I can fix this problem on my own, either by modifying the scalars to be wrapped in an array or by modifying some functions of the library itself. I eventually did use both of these strategies to get my case to work.

It just seemed like the process was more difficult than necessary, as I had to keep checking that my array shapes were exactly meet to those expected by the code, when there wasn't a lot of guidance on what those necessary array shapes are. 

Regardless, the library is very useful and great for relatively quick implementations of Kalman filtering problems! Thank you for putting it together!
